# noMisleadingCharacterClass

Disallow characters made with multiple code points in character class syntax.

## Diagnostic Category: `lint/suspicious/noMisleadingCharacterClass`

### Since: `v1.5.0`

This rule is recommended by Biome. A diagnostic error will appear when linting your code.
This rule has a **safe** fix.

Sources: 
- Same as: no-misleading-character-class

Disallow characters made with multiple code points in character class syntax.

Unicode includes the characters which are made with multiple code points. e.g. AÌ, ğŸ‡¯ğŸ‡µ, ğŸ‘¨â€ğŸ‘©â€ğŸ‘¦.
A RegExp character class `/[abc]/` cannot handle characters with multiple code points.
For example, the character `â‡ï¸` consists of two code points: `â‡` (U+2747) and `VARIATION SELECTOR-16` (U+FE0F).
If this character is in a RegExp character class, it will match to either `â‡` or `VARIATION SELECTOR-16` rather than `â‡ï¸`.
This rule reports the regular expressions which include multiple code point characters in character class syntax.

## Examples

### Invalid

```js
/^[AÌ]$/u;
```

```js
/^[â‡ï¸]$/u;
```

```js
/^[ğŸ‘¶ğŸ»]$/u;
```

```js
/^[ğŸ‡¯ğŸ‡µ]$/u;
```

```js
/^[ğŸ‘¨â€ğŸ‘©â€ğŸ‘¦]$/u;
```

```js
/^[ğŸ‘]$/; // surrogate pair without u flag
```

### Valid

```js
/^[abc]$/;
/^[ğŸ‘]$/u;
/^[\q{ğŸ‘¶ğŸ»}]$/v;
```

## Related links

- Disable a rule
- Configure the rule fix
- Rule options